core/templates/dev/head/pages/exploration_editor/simple_editor_tab/sidebarHighlightingService.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that highlights the color of the question
+ * depending on wheather it is finished or unfinished .
+ */
+
+oppia.factory('SidebarHighlightingService', [
+  function() {
+    var ERROR_SELECTOR = 'simple-editor-sidebar .error';
+    var ERROR_CLASS = 'error';
+
+    return {
+      errorHighlight: function(id) {
+          if (!angular.isDefined(id)) {
+            return;
+          } else {
+            var elm = angular.element(document.getElementById(
+              QuestionIdService.SIDEBAR_PREFIX + id)
+            );
+            elm.addClass(ERROR_CLASS);
+          }
+        },
+      undoErrorHighlight: function(id) {
```

Comment:
This method takes and id, but it does not use it, it removes the 'error' class from all sidebar items that have it, i think when using this function we will probably want to undo only one.